dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityChangeLogServiceTest.java

```diff
@@ -338,14 +373,39 @@ void shouldReturnChangeLogsFromAccessibleTEAWhenProgramNotSpecified() throws Not
 
     Page<TrackedEntityChangeLog> changeLogs =
         trackedEntityChangeLogService.getTrackedEntityChangeLog(
-            UID.of(trackedEntity.getUid()), null, defaultPageParams);
+            UID.of(trackedEntity.getUid()), null, defaultOperationParams, defaultPageParams);
 
     assertNumberOfChanges(1, changeLogs.getItems());
     assertAll(
         () ->
             assertCreate(
-                trackedEntityAttribute.getUid(), "new value", changeLogs.getItems().get(0)),
-        () -> assertUser(user, changeLogs.getItems().get(0)));
+                trackedEntityAttribute.getUid(), "new value", changeLogs.getItems().get(0)));
+  }
+
+  @Test
+  void shouldReturnChangeLogsFromSpecifiedProgramOnlyWhenMultipleLogsExist()
+      throws NotFoundException {
+    updateAttributeValue(trackedEntityAttributeValue, ChangeLogType.CREATE, "new value");
+    updateAttributeValue(outOfScopeTrackedEntityAttributeValue, ChangeLogType.CREATE, "new value");
+    Program program = createAndAddProgram('C');
+
+    programOwnerService.createOrUpdateTrackedEntityProgramOwner(
+        trackedEntity.getUid(), program.getUid(), orgUnitA.getUid());
+
+    createAndPersistProgramAttribute(program);
```

Comment:
I don't get how the program influences the result. What attribute is a program attribute in here? Can we adapt the helper to make that clear and remove other things from the setup here that are just details?

Reply:
I'm creating values for two different TEA. 
The program influences the result because only one of these two TEA belong to this program, hence the name starting with "outOfScope" for the second TEA.
In this case, only the first TEA is a program attribute, because I'm creating it at the line you pinned.

The thing with moving something to the setup is it will affect other tests, and we don't want to create a program attribute or set the ownership of a program in all the tests.